netgrasp/utils/exclusive_lock.py

```diff
@@ -2,64 +2,65 @@
 import os
 import errno
 import time
-from netgrasp.utils import debug
+
 from netgrasp.utils import simple_timer
 
-class ExclusiveFileLock:
-    def __init__(self, ng, timeout, name):
-        self.ng = ng
-        self._lockfile = ng.database["lock"]
+
+class ExclusiveFileLock(object):
+
+    def __init__(self, lockfile, timeout, name, debugger):
```

Comment:
My reasoning for not passing this information in is it makes calling ExclusiveFileLock simpler; you don't have to remember the flags and all the parameters to pass in. We already have access to all this information from the ng object, so why complicate things? (If you go back through history, you'll see we used to pass in all these parameters). So, unless there's a very good reason for needing to change back again I have no intent of merging this change.